packages/aws-cdk/lib/api/hotswap/lambda-functions.ts

```diff
@@ -110,10 +166,50 @@ class LambdaFunctionHotswapOperation implements HotswapOperation {
   }
 
   public async apply(sdk: ISDK): Promise<any> {
-    return sdk.lambda().updateFunctionCode({
-      FunctionName: this.lambdaFunctionResource.physicalName,
-      S3Bucket: this.lambdaFunctionResource.code.s3Bucket,
-      S3Key: this.lambdaFunctionResource.code.s3Key,
-    }).promise();
+    const lambda = sdk.lambda();
+    const resource = this.lambdaFunctionResource.resource;
+
+    if (resource.code !== undefined) {
+      await lambda.updateFunctionCode({
+        FunctionName: this.lambdaFunctionResource.physicalName,
+        S3Bucket: resource.code.s3Bucket,
+        S3Key: resource.code.s3Key,
+      }).promise();
+    }
+
+    if (resource.tags !== undefined) {
+      // It's really unfortunate, but Lambda allows function _names_ for code
+      // updates, but we need the function ARN if we're managing tags. We also
+      // can't assume we'll get it from the preceding code update, since there
+      // might not _be_ a code update.
+      const tagsToDelete = Object.entries(resource.tags!.tagUpdates)
```

Comment:
Can you specify the type of `tagsToDelete` explicitly here, like you do for `tagsToSet`?